clientconn.go

```diff
@@ -1322,24 +1322,36 @@ func (ac *addrConn) createTransport(addr resolver.Address, copts transport.Conne
 	}
 
 	select {
-	case <-time.After(time.Until(connectDeadline)):
+	case <-connectCtx.Done():
 		// We didn't get the preface in time.
-		err := fmt.Errorf("failed to receive server preface within timeout")
-		newTr.Close(err)
-		channelz.Warningf(logger, ac.channelzID, "grpc: addrConn.createTransport failed to connect to %v: %v", addr, err)
-		return err
+		newTr.Close(transport.ErrConnClosing)
```

Comment:
Let's make it clear that the use of `transport.ErrConnClosing` isn't important? Otherwise later we may think this error value means something.

Reply:
Oopsie ... race !!